packages/plugin-workspace-tools/sources/commands/foreach.ts

```diff
@@ -54,11 +57,20 @@ export default (clipanion: any, pluginConfiguration: PluginConfiguration) => cli
   }))
 
   .action(
-    async ({cwd, args, stdout, command, exclude, include, interlaced, parallel, topological, topologicalDev, verbose, jobs, ...env}: ForeachOptions) => {
+    async ({cwd, args, stdout, command, exclude, include, interlaced, parallel, topological, topologicalDev, all, verbose, jobs, ...env}: ForeachOptions) => {
       const configuration = await Configuration.find(cwd, pluginConfiguration);
-      const {project} = await Project.find(configuration, cwd);
+      const {project, workspace} = await Project.find(configuration, cwd);
+
+      const targetCwds: Set<PortablePath> = new Set<PortablePath>(
+        all ? project.workspaces.map(workspace => workspace.cwd) : workspace ? workspace.workspacesCwds : []
```

Comment:
The double ternary is a bit hard to read 🤔

The logic doesn't support nested worktrees, is that expected? Something like this:

```
A - B
 \
  C - D
   \
    E
```

In this case, running `foreach` in `A` will not run it in `D` and `E`.

Reply:
@arcanis Good point, I will fix this